packages/common/src/ContractUtils.js

```diff
@@ -27,6 +27,61 @@ const revertWrapper = result => {
   return result;
 };
 
+/**
+ * Simulate transaction via .call() and then .send() and return receipt. If an error is thrown,
+ * return the error and add a flag denoting whether it was sent on the .call() or the .send().
+ * @param {*Object} transaction Transaction to call `.call()` and subsequently `.send()` on from `senderAccount`.
+ * @param {*String} senderAccount Sender of transaction. Injected to `transactionConfig`.
+ * @param {*Object} transactionConfig { gasPrice } passed to web3 transaction.
+ * @param {*Number} transactionGasLimit Maximum gas limit to set for sending transaction.
+ * @param {*Number} gasLimitBuffer Multiplier applied to Truffle's estimated gas limit for a transaction to send.
+ * @return Error and type of error (originating from `.call()` or `.send()`) or transaction receipt and return value.
+ */
+const runTransaction = async (
+  transaction,
+  senderAccount,
+  transactionConfig,
+  transactionGasLimit,
```

Comment:
Thoughts on shortening these variable names?
```suggestion
  sender,
  config,
  gasLimit,
```

Reply:
Done